sds_data_manager/stacks/ialirt_processing_stack.py

```diff
@@ -0,0 +1,63 @@
+"""Processing Stack
+This is the module containing the general stack to be built for
+computation of I-ALiRT algorithms
+"""
+from aws_cdk import Stack
+from aws_cdk import aws_ec2 as ec2
+from aws_cdk import aws_ecr as ecr
+from constructs import Construct
+
+from sds_data_manager.constructs.ialirt_compute_resources import (
+    IalirtEC2Resources,
+)
+from sds_data_manager.stacks import ialirt_dynamodb_stack
+
+
+class IalirtProcessing(Stack):
+    """A processing system for ialirt."""
+
+    def __init__(
+        self,
+        scope: Construct,
+        construct_id: str,
+        vpc: ec2.Vpc,
+        repo: ecr.Repository,
+        **kwargs,
+    ) -> None:
+        """Constructor
+
+        Parameters
+        ----------
+        scope : Construct
+            Parent construct.
+        construct_id : str
+            A unique string identifier for this construct.
+        vpc : ec2.Vpc
+            VPC into which to put the resources that require networking.
+        repo : ecr.Repository
+            ECR repository containing the Docker image.
+        """
+        super().__init__(scope, construct_id, **kwargs)
+
+        # Setup the EC2 resources
+        self.ec2_resources = IalirtEC2Resources(
+            self,
+            "IalirtEC2Environment",
+            vpc=vpc,
+            repo=repo,
+        )
+
+        # I-ALiRT IOIS DynamoDB
+        # ingest-ugps: ingestion ugps - 64 bit
+        # sct-vtcw: spacecraft time ugps - 64 bit
+        # src-seq-ctr: increments with each packet (included in filename?)
+        # ccsds-filename: filename of the packet
+        ialirt_dynamodb_stack.DynamoDB(
+            scope,
+            construct_id="IalirtDynamoDB",
+            table_name="ialirt-iois",
+            partition_key="ingest-ugps",
+            sort_key="sct-vtcw",
```

Comment:
I think we should make these more descriptive names for us. The iois team isn't going to use this table, you'll be the one interacting with it.

```suggestion
            # Are we storing individual packets in this table, or the binary file delivered every minute?
            table_name="ialirt-packets",
            partition_key="ingest-time",
            sort_key="spacecraft-time",
```

Reply:
Is the plan to have all queries be ingest-time = X and spacecraft_time between Y and Z? Is there ever a need to query by a range of ingest-time values?

Reply:
That's the plan. Data will be ingested every minute (ingest time) and spacecraft_time will be the actual timestamp of the packets. I do not anticipate a need to query the ingest-time values. They are only to be used for populating the database.

Reply:
Ok, just to clarify, will the ingest_time be known and provided in every query?

With Dynamo we cannot query the sort_key (spacecraft_time) without specifying the partition_key (ingest_time).

Reply:
Yes. The ingest_time will be known and provided in every query. Thank you for the heads up.

Reply:
> Yes. The ingest_time will be known and provided in every query.

Maybe this is a naming thing, but what is `ingest_time`? I'm not sure I'd want to keep track of that to make a query. My use-case would be the following query: "Give me all packets between 2023-01-01T00:00 and 2023-01-02T00:00". I don't necessarily know when those packets were ingested I don't think.

Reply:
I think we might be getting ahead of ourselves with the database conversation. As I mentioned in the Overview of this PR,  I expect the setup of the database to change so try not to focus too much on that part of the code. The reason I said that is that I want to reevaluate whether or not we even want to put metadata from the CCSDS packets into the database if we are already storing them in S3. It might make sense to decom them first and then put the decom'd information in the database instead. So it is still debatable whether or not we even need it for ingest. But if we do use it for ingest then we will need a primary key and sort key of which the combination is unique. Since OIS is delivering the packets on a minute cadence to me it makes sense to have a primary key that is ingest time. But I am not opposed to changing that.